include/circt/Dialect/RTL/RTLStructure.td

```diff
@@ -138,6 +138,102 @@ def RTLModuleExternOp : RTLOp<"module.extern",
   let parser = "return ::parse$cppClass(parser, result);";
 }
 
+def RTLGeneratorTypeOp : RTLOp<"generator.type", 
+      [Symbol, HasParent<"mlir::ModuleOp">]> {
+  let summary = "RTL Generator Type";
+  let description = [{
+    the "rtl.generator.type" operation declares a type of generated module.  The
```

Comment:
The word "type" here is very confusing to me.  I think that this can just be "rtl.generator" without any loss, right?

This if you want another word in there, you could say that this is a declaration or that this is a module generator, e.g. "rtl.generator.module" -> generator of modules.